foundation/org.eclipse.persistence.core/pom.xml

```diff
@@ -37,53 +37,66 @@
-                                <filter token="BUILD_REVISION"  value="${svn.revision}"/>
-                                <copy file="${src}/org/eclipse/persistence/Version.java" tofile="${src}/org/eclipse/persistence/Version.j"    filtering="false"/>
-                                <copy file="${src}/org/eclipse/persistence/Version.j"    tofile="${src}/org/eclipse/persistence/Version.java" filtering="true"/>
-                            </target>
-                        </configuration>
+                        <id>build.date</id>
+                        <phase>initialize</phase>
                         <goals>
-                          <goal>run</goal>
+                            <goal>create-timestamp</goal>
                         </goals>
+                        <configuration>
+                            <timestampFormat>yyyyMMdd</timestampFormat>
+                            <timestampPropertyName>build.date</timestampPropertyName>
+                        </configuration>
                     </execution>
                     <execution>
-                        <id>cleanup-stamp-version</id>
-                        <phase>package</phase>
+                        <id>build.time</id>
+                        <phase>initialize</phase>
+                        <goals>
+                            <goal>create-timestamp</goal>
+                        </goals>
                         <configuration>
-                            <target>
-                                <echo message="Clean-up after stamping."/>
-                                <copy file="${src}/org/eclipse/persistence/Version.j"    tofile="${src}/org/eclipse/persistence/Version.java" filtering="false" overwrite="true"/>
-                                <delete file="${src}/org/eclipse/persistence/Version.j"/>
-                            </target>
+                            <timestampFormat>HHmm</timestampFormat>
+                            <timestampPropertyName>build.time</timestampPropertyName>
                         </configuration>
+                    </execution>
+                    <execution>
+                        <phase>initialize</phase>
+                        <goals>
+                            <goal>create</goal>
+                        </goals>
+                    </execution>
+                </executions>
+            </plugin>
+            <!--Prepare version.properties file. Store there info about version, build date/time, git commit id-->
+            <plugin>
+                <artifactId>maven-resources-plugin</artifactId>
+                <version>3.1.0</version>
+                <executions>
+                    <execution>
+                        <id>copy-resources</id>
+                        <phase>process-resources</phase>
                         <goals>
-                            <goal>run</goal>
+                            <goal>copy-resources</goal>
```

Comment:
Is this whole plugin config required? I'd assume version.properties file will be put under $resources folder and resources configuration would be updated to enable filtering on this particular resource folder (see https://maven.apache.org/plugins/maven-resources-plugin/examples/filter.html).